testing/internal/e2e/tests/database/migration_test.go

```diff
@@ -0,0 +1,226 @@
+	t.Cleanup(func() {
+		require.NoError(t, newNetwork.Remove(ctx))
+	})
+
+	// Create a Boundary database
+	databaseC := docker.CreateBoundaryDatabase(t, networkName)
+	t.Cleanup(func() {
+		require.NoError(t, databaseC.Terminate(ctx))
+	})
+
+	// Create a target
+	targetC := docker.CreateTarget(t, c.TargetSshUser, c.TargetSshKeyPath, networkName)
+	t.Cleanup(func() {
+		require.NoError(t, targetC.Terminate(ctx))
+	})
+
+	// Initialize the database
+	dbInitC := docker.InitBoundaryDatabase(t, databaseC.UriNetwork, networkName)
+	t.Cleanup(func() {
+		require.NoError(t, dbInitC.Terminate(ctx))
+	})
+
+	dbInitLogs, err := dbInitC.Logs(ctx)
+	require.NoError(t, err)
+	dbInitData, err := io.ReadAll(dbInitLogs)
+	require.NoError(t, err)
+
+	type AuthMethodInfo struct {
+		AuthMethodId string `json:"auth_method_id"`
+		LoginName    string `json:"login_name"`
+		Password     string `json:"password"`
+	}
+
+	type DbInitInfo struct {
+		AuthMethod AuthMethodInfo `json:"auth_method"`
+	}
+
+	var dbInitInfo DbInitInfo
+	err = json.Unmarshal(dbInitData, &dbInitInfo)
+	require.NoError(t, err, fmt.Sprintf("Error parsing: %v", dbInitData))
+
+	// Start a Boundary server
+	boundaryC := docker.StartBoundary(t, databaseC.UriNetwork, networkName)
+	t.Cleanup(func() {
+		require.NoError(t, boundaryC.Terminate(ctx))
+	})
+
+	// Populate database with additional data
+	os.Setenv("BOUNDARY_ADDR", boundaryC.UriLocalhost)
+	os.Setenv("E2E_PASSWORD_AUTH_METHOD_ID", dbInitInfo.AuthMethod.AuthMethodId)
+	os.Setenv("E2E_PASSWORD_ADMIN_LOGIN_NAME", dbInitInfo.AuthMethod.LoginName)
+	os.Setenv("E2E_PASSWORD_ADMIN_PASSWORD", dbInitInfo.AuthMethod.Password)
+
+	boundary.AuthenticateAdminCli(t, ctx)
+	newOrgId := boundary.CreateNewOrgCli(t, ctx)
+	newProjectId := boundary.CreateNewProjectCli(t, ctx, newOrgId)
+	newHostCatalogId := boundary.CreateNewHostCatalogCli(t, ctx, newProjectId)
+	newHostSetId := boundary.CreateNewHostSetCli(t, ctx, newHostCatalogId)
+	newHostId := boundary.CreateNewHostCli(t, ctx, newHostCatalogId, targetC.UriNetwork)
+	boundary.AddHostToHostSetCli(t, ctx, newHostSetId, newHostId)
+	newTargetId := boundary.CreateNewTargetCli(t, ctx, newProjectId, "2222")
+	boundary.AddHostSourceToTargetCli(t, ctx, newTargetId, newHostSetId)
+
+	newAwsHostCatalogId := boundary.CreateNewAwsHostCatalogCli(t, ctx, newProjectId, c.AwsAccessKeyId, c.AwsSecretAccessKey)
+	newAwsHostSetId := boundary.CreateNewAwsHostSetCli(t, ctx, newAwsHostCatalogId, c.AwsHostSetFilter)
+	boundary.WaitForHostsInHostSetCli(t, ctx, newAwsHostSetId)
+
+	newCredentialStoreId := boundary.CreateNewCredentialStoreStaticCli(t, ctx, newProjectId)
+	boundary.CreateNewStaticCredentialPasswordCli(t, ctx, newCredentialStoreId, c.TargetSshUser, "password")
+	jsonCredentialFilePath := fmt.Sprintf("%s/%s", t.TempDir(), "test.json")
+	f, err := os.Create(jsonCredentialFilePath)
+	require.NoError(t, err)
+	_, err = f.WriteString(fmt.Sprintf("{ \"a\": \"test\", \"b\": \"foo\" }\n"))
+	require.NoError(t, err)
```

Comment:
Does this file need to be written out each time? I think it could just be added to `testdata` and used from there.